pkg/agent/clean/clean.go

```diff
@@ -0,0 +1,325 @@
+				}
+			}
+
+			// Cleanup annotations
+			for _, anno := range nsAnnotations {
+				if _, ok := nameSpace.Annotations[anno]; ok {
+					updated = ok
+					delete(nameSpace.Annotations, anno)
+				}
+			}
+
+			if updated {
+				_, err = client.CoreV1().Namespaces().Update(nameSpace)
+				if err != nil {
+					return err
+				}
+			}
+
+			return nil
+		})
+
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	return errs
+
+}
+
+func cleanupClusterRoleBindings(client *kubernetes.Clientset) []error {
+	logrus.Info("Starting cleanup of clusterRoleBindings")
+	crbs, err := client.RbacV1().ClusterRoleBindings().List(listOptions)
+	if err != nil {
+		return []error{err}
+	}
+
+	var errs []error
+
+	for _, crb := range crbs.Items {
+		err = client.RbacV1().ClusterRoleBindings().Delete(crb.Name, &metav1.DeleteOptions{})
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+	return errs
+}
+
+func cleanupRoleBindings(client *kubernetes.Clientset) []error {
+	logrus.Info("Starting cleanup of roleBindings")
+	rbs, err := client.RbacV1().RoleBindings("").List(listOptions)
+	if err != nil {
+		return []error{err}
+	}
+
+	var errs []error
+
+	for _, rb := range rbs.Items {
+		err = client.RbacV1().RoleBindings(rb.Namespace).Delete(rb.Name, &metav1.DeleteOptions{})
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+	return errs
+}
+
+func cleanupClusterRoles(client *kubernetes.Clientset) []error {
+	logrus.Info("Starting cleanup of clusterRoles")
+	crs, err := client.RbacV1().ClusterRoles().List(listOptions)
+	if err != nil {
+		return []error{err}
+	}
+
+	var errs []error
+
+	for _, cr := range crs.Items {
+		err = client.RbacV1().ClusterRoles().Delete(cr.Name, &metav1.DeleteOptions{})
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+	return errs
+}
+
+func cleanupRoles(client *kubernetes.Clientset) []error {
+	logrus.Info("Starting cleanup of roles")
+	rs, err := client.RbacV1().Roles("").List(listOptions)
+	if err != nil {
+		return []error{err}
+	}
+
+	var errs []error
+
+	for _, r := range rs.Items {
+		err = client.RbacV1().Roles(r.Namespace).Delete(r.Name, &metav1.DeleteOptions{})
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+	return errs
+}
+
+func deleteJob(client *kubernetes.Clientset) error {
+	logrus.Info("Starting cleanup of jobs")
+	jobs, err := client.BatchV1().Jobs("default").List(listOptions)
+	if err != nil {
+		return err
+	}
+
+	for _, job := range jobs.Items {
+		prop := metav1.DeletePropagationBackground
+		if strings.HasPrefix(job.Name, "cattle-cleanup") {
+			err = client.BatchV1().Jobs("default").Delete(job.Name, &metav1.DeleteOptions{
+				PropagationPolicy: &prop,
+			})
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// tryUpdate runs the input func and if the error returned is a conflict error
+// from k8s it will sleep and attempt to run the func again. This is useful
+// when attempting to update an object.
+func tryUpdate(f func() error) error {
+	timeout := 100
```

Comment:
I think this variable is improperly named. It isnt a timeout, its a sleep or a wait

Reply:
Updated